pkg/command/context.go

```diff
@@ -273,20 +275,30 @@ func getPromptOpts() []component.PromptOpt {
 }
 
 func createNewContext() (context *configtypes.Context, err error) {
-	ctxCreationType := ""
+	ctxCreationType := ContextUnknown
 	contextType = strings.TrimSpace(contextType)
```

Comment:
For a better UX, should we check the validity of the `contextType` value which comes from the `--type` flag?

If we don't and the users make an mistake in the `--type`, we can end up with slightly confusing situations such as:
```
# Notice the misspelled "kubenetes"
tz context create --type kubenetes
? Select context creation type  [Use arrows to move, type to filter]
> Mission Control
  Application Engine
  Kubernetes (Cluster Endpoint)
  Kubernetes (Local Kubeconfig)
```

Reply:
Validation of the `--type` can be done using `IsValidContextType` API https://github.com/vmware-tanzu/tanzu-plugin-runtime/pull/112/files#diff-a51ca6e4ebbf11582e078a669e21c58bb8937b2840464358a059a7017314e63aR48 API.

I have been using it my other PR. https://github.com/vmware-tanzu/tanzu-cli/pull/524/files#diff-f1ed1c998c6b28633106cbbfd6a643f1efede45708937540d4551397adeef1fdR753

So, I can add the validation to the `tanzu context create --type` command as well as part of my PR once this PR gets in.